types/user.ts

```diff
@@ -111,8 +111,8 @@ export class User extends LazyObject {
 
     if (!content.hasData) await content.getData();
 
-    // if already reported, return
-    if (await this.alreadyReported(content)) return;
+    // if already reported or already has a significant number of reports (arbitrary), return
+    if (await this.alreadyReported(content) || content.reports?.length! > 7) return;
 
     // update client side
     content.reports?.push(this);
```

Comment:
actually, users should think their report is going through even if there are already 7. Should we update their clientside values of the reports array and let it be out of sync until the next get, or should frontend handle it by making the report flag icon turn red anyways (or whatever visual feedback we choose)? @tylerhm

Reply:
i mean they shouldn't really be able to see it if there's that many reports... so it's really just an issue of checking it within the transaction if we're doing that

Reply:
This discussion can be moved to the issue that was filed, since it's not actually blocking this PR being merged

Reply:
right, users shouldn't be able to tell. I think we can go ahead with the current solution and change it later if we don't like it